packages/allure-js-commons/src/writers/AllureWriter.ts

```diff
@@ -6,7 +6,7 @@ export interface AllureWriter {
 
   writeGroup(result: TestResultContainer): void;
 
-  writeAttachment(name: string, content: Buffer | string): void;
+  writeAttachment(name: string, content: Buffer | string, encoding?: BufferEncoding): void;
```

Comment:
Not all usages require the encoding, so it is an optional parameter in the interface.